### ISSUE TITLE
Port XQF to GTK+3

### ISSUE THREAD

**Issue by illwieckz** [XQF/xqf#14]
This issue must be closed before :
- Fix XQF to have an undeprecated and clean GTK+2 version #13

**Comment by illwieckz** [comment]
@skybon said:

> The process is described at https://developer.gnome.org/gtk3/stable/gtk-migrating-2-to-3.html
> Depends on: #92

**Comment by illwieckz** [comment]
#92 is **Avoid individual GLib/GTK includes**

**Comment by Xylemon** [comment]
If you do this can you please keep the option for users to compile a GTK2 version?

**Comment by illwieckz** [comment]
@skybon, @jmallach , do you think @Xylemon proposal is achievable without much effort or not? (my own opinion is _“the smaller the code base is, the best it is”_).

**Comment by skybon** [comment]
@illwieckz I have not looked at GTK3 yet so cannot say for sure.

However, I cannot imagine a strong case for GTK2 since GTK3 is the default in nigh all LTS distributions now.

**Comment by illwieckz** [comment]
@skybon's PR #147 laid the first GTK3 stone

**Comment by jmallach** [comment]
On Wed, Jun 10, 2015 at 02:08:19PM -0700, Thomas Debesse wrote:
> @skybon, @jmallach , do you think @Xylemon proposal is achievable without much effort or not? (my own opinion is _“the smaller the code base is, the best it is”_).


@Xylemon, do you have a good reason to want to keep a GTK2 option? GTK+3
should be available on any distro these days, and the list of benefits is so
long I wouldn't know where to start: HiDPI, Wayland, etc...

Unless there's really a good reason to keep GTK+2 as an option, I would
personally be against it.

-- 
Jordi Mallach Pérez  --  Debian developer     http://www.debian.org/
jordi@sindominio.net     jordi@debian.org     http://www.sindominio.net/
GnuPG public key information available at http://oskuro.net/

**Comment by illwieckz** [comment]
I agree with @jmallach, unless someone comes with an unattended strong reason, I see no need for keeping GTK+2 if we have GTK+3.

**Comment by Xylemon** [comment]
There are many reasons people would not want to move to GTK3. Some of them are political, it being an out of place program on their systems, and just the general overhead of features that I don't think XQF would ever even utilize to be brutally honest. How would Wayland and HiDPI even be useful? You'd have to do a lot of work in the first place (such as vector images and so on). Not to mention if you really want to support distributions like Slackware, most users aren't going to want to compile a GTK3 only program. In fact almost all SlackBuilds are set up to use GTK2 because the majority of them use KDE or some type of GTK2 system. I still don't see why it would be such a big deal to support GTK2 and GTK3 simultaneously, I mean projects like SpaceFM handle this very well. But if you guys desire to do such a move then I suppose there's nothing I can do.

**Comment by skybon** [comment]
@Xylemon Current XQF code is barely maintainable and all substantial improvements will be coupled with rewrite. Eventually when GTK3 version is feature complete old parts will be scrapped.

*Political reasons* (IDONTLIKEIT) cannot substitute the fact that GTK2 is nigh dead. It does not receive any fixes except for security. GTK3 is GNOME's focus and therefore the version to be used by projects.

If anyone of Slackware is willing to step up and do GTK2 backporting then we would have something to talk about. Otherwise it's an additional maintenance burden that I'm very much against.

**Comment by illwieckz** [comment]
@aufau reported in #222 that GTK2-based XQF does not support desktop scaling, and I reproduced the issue and confirmed that restarting the app after having rescaled it does not fix it. I assume this is a GTK2 issue and the proper fix is to port to GTK3.

![xqf-scaling](https://user-images.githubusercontent.com/4474909/62724032-5d5c0480-ba12-11e9-9060-52758b024358.png)

**Comment by jmallach** [comment]
I played with Claude Code to push the current state of the porting to completion. There are rough edges and lots of polish to do, but it works. You can see the current state in https://github.com/jmallach/xqf/tree/gtk4.

If the use of Claude is acceptable for this project, we can work to finish this port and finally commit it. If it's not, I guess it will go to the wastebasket.

**Comment by illwieckz** [comment]
Hi @jmallach! I just want to say I'm sorry as your mail went into my spam folder, and I discovered it almost a month late, and then I had been very busy and didn't found time to answer it, but I wanted to answer it.

To me this kind of code translation work is fine for AI. I would prefer we avoid AI but that's one rare case where it's clear the AI is _assisting_. And this is an API translation so there is nothing really original in such work.

One very important thing though: Agents have no accountability. So the people committing the code assisted by AI should take the full accountability of the contribution.

That means that no commit should ever use any `Co-authored-by:` for a AI agent.

Disclosing the use of an agent is fine, but then the tag should be `Assisted-by:`, not `Co-authored-by:`. That's because the AI agent is not an author and should not be (it has no accountability, and people should not be subjected through copyright laws to a thing that has no accountability and that cannot die).

**Comment by illwieckz** [comment]
Rewriting the tag in your commits can very likely be scripted, if you need help on that just tell me. 🙂️

**Comment by jmallach** [comment]
Yeah, that will be no problem.

**Comment by jmallach** [comment]
I've pushed everything again. If you want, I can open the PR so you can have a look at it.

**Comment by illwieckz** [comment]
Yes please open a PR!

I noticed you used `Assisted-By:` instead of `Assisted-by:` (with a major `b` instead of a minor `b`). I don't know if the case is important for tooling handling them, but better use the minor b which is the convention.

**Comment by jmallach** [comment]
Doh that was a silly typo. Fixed too. I'll open a PR.
